src/Tools/MicroBuild/publish-assets.ps1

```diff
@@ -1,136 +1,193 @@
+# Publish a given entry: branch or release. 
+function Publish-Entry($publishData, [switch]$isBranch) { 
+    Test-Entry $publishData -isBranch:$isBranch
+    $packageDir = Join-Path $nugetDir $publishData.nugetKind
 
-    # Load in the NuGet.exe information
-    . "$PSScriptRoot\..\..\..\build\scripts\LoadNuGetInfo.ps1"
-    write-host "NuGet.exe path is $nugetexe"
 
-    $exitCode = 0
+    # First publish the NuGet packages to the specified feeds
+    foreach ($url in $publishData.nuget) { 
+        Publish-NuGet $packageDir $url
+    }
 
-    Write-Host "Uploading NuGet packages..."
+    # Next publish the VSIX to the specified feeds
+    $vsixData = $publishData.vsix
+    if ($vsixData -ne $null) { 
+        Publish-Vsix $vsixData
+    }
 
-    $nugetPath = Join-Path $binariesPath "NuGet\PerBuildPreRelease"
+    # Finally get our channels uploaded to versions
+    foreach ($channel in $publishData.channels) {
+        Publish-Channel $packageDir $channel
+    }
 
-    [xml]$packages = Get-Content "$nugetPath\myget_org-packages.config"
+    exit 0
+}
 
-    $sourceUrl = "https://dotnet.myget.org/F/roslyn/api/v2/package"
-    
-    pushd $nugetPath
-    foreach ($package in $packages.packages.package)
-    {
-        $nupkg = $package.id + "." + $package.version + ".nupkg"
-        Write-Host "  Uploading '$nupkg' to '$sourceUrl'"
-        if (-not (test-path $nupkg))
-        {
-            Write-Error "NuGet $nupkg does not exist"
-            $exitCode = 6
-        }
+function Test-Member($obj, [string]$name) { 
+    $value = Get-Member -Name $name -InputObject $obj 
+    return $value -ne $null
+}
 
-        if (-not $test) 
-        {
-            & "$NuGetExe" push "$nupkg" `
-                -Source $sourceUrl `
-                -ApiKey $apiKey `
-                -NonInteractive `
-                -Verbosity quiet
-            if ($LastExitCode -ne 0)
-            {
-                Write-Error "Failed to upload NuGet package: $nupkg"
-                $exitCode = 3
-            }
-        }
+# This script is interested in the short branch name: master, dev15.x, etc ... But several
+# of our publish operations specify fully branch names like /refs/heads/master. Normalizing
+# those out to the short branch name here.
+function Normalize-BranchName([string]$branchName) {
+    switch -regex ($branchName) { 
+        "refs/heads/(.*)" { return $matches[1] }
+        "refs/pull/\d*/(.*)" { return $matches[1] }
+        default { return $branchName }
     }
-    popd
+}
 
-    Write-Host "Uploading VSIX extensions..."
+try {
+    . (Join-Path $PSScriptRoot "..\..\..\build\scripts\build-utils.ps1")
+    $nuget = Ensure-NuGet
+    $nugetDir = Join-Path $configDir "NuGet"
 
-    $vsixPath = $binariesPath
+    if ($configDir -eq "") {
+        Write-Host "Must provide the build output with -configDir"
+        exit 1
+    }
 
-    [xml]$extensions = Get-Content "$vsixPath\myget_org-extensions.config"
+    # TODO: use the live file 
```

Comment:
This TODO will be fixed as soon as I commit this PR. I need to have this commited so I can then download it from master.